src/lambdas/tests/random-get.test.ts

```diff
@@ -0,0 +1,44 @@
+import fetch from 'node-fetch'
+import { randomGet } from '../controller'
+import { mockOne, mockOneResult } from './mock'
+
+const mockedFetch: jest.Mock = fetch as any
+
+jest.mock('node-fetch')
+
+describe('random-get handler', () => {
+  it('returns payload from fetch request', async () => {
+    mockedFetch.mockReturnValueOnce(
+      Promise.resolve({
+        status: 200,
+        json: () => Promise.resolve(mockOne),
+      }),
+    )
+    const response = await randomGet()
+    expect(response).toMatchObject({ body: mockOneResult, statusCode: 200 })
+  })
+
+  it('returns status 500 if breed api returns status code >= 400', async () => {
+    mockedFetch.mockReturnValueOnce(
+      Promise.resolve({
+        status: 401,
+        json: () => Promise.resolve({ message: 'Unauthorized' }),
+      }),
+    )
+    const response = await randomGet()
+    expect(response).toMatchObject({
+      message: 'Failed to get all Breeds',
+      statusCode: 500,
+      errorCode: '00001',
+    })
+  })
+
+  it('returns status 500 if breed api rejects', async () => {
+    mockedFetch.mockReturnValueOnce(Promise.reject(new Error('TIMEOUT')))
+    const response = await randomGet()
+    expect(response).toMatchObject({
+      message: 'General Error occured getting all breeds',
+      statusCode: 500,
+    })
+  })
```

Comment:
I really appreciate that you have two different tests here, one general and one that is status codd specific.  

One of the key components of this challenge is an accurate timeout test.  So we would need to consider, how `fetch` times out and what that resulting error looks like and then write a test that only tests for that singular timeout use case. 

Please let me know if you want some more clarification with this one.

Reply:
I might need more clarification. I added an additional test to handle a timeout error by setting a very low timeout ms. Please let me know if this is what you are asking for.

Reply:
For sure...

So what we are fishing for: 
- what would cause fetch to timeout?
- what type of error would that timeout produce?
- how can we test for that specific error?

So rather than setting up for a race condition versus a timeout that we may or may not hit, instead consider how we can mock fetch to return the expected error and then test for that error. 

Hopefully, that clarifies things.  But please let me know if not and I am happy to try again.  

Cheers

Reply:
Thanks for clarification. I mock rejected with the correct timeout error.